### DISCUSSION TITLE
关于包重复的问题

### DISCUSSION THREAD

**Discussion by Extra-Creativity** [xmake-io/xmake#3293] [Q&A] [answered]
我很久之前写的一个project，里面有`add_requires()`这个部分；更新xmake之后，重新执行这个工程，它又把一些包都安装了一遍。

然后，我查看`~/.xmake/packages/{相应文件夹}`，发现同一版本的包出现了两份（哈希值不一样）；然后我用`xrepo remove 这个包`，发现只删除了一份。

我想问这是什么情况？包的版本没变，是xmake更新导致换源了吗？有什么办法可以在latest包版本没变的情况下不重新下载呢？此外，有没有直接卸载全部同版本重复包的方法？

**Comment by waruqi** [comment, marked as answer]
应该是你之前用的很老的版本安装的包，中间有几个版本 更新过 hash 计算，但也是很久之前的事了，现在应该已经稳定了

没啥特殊问题，不会轻易变动，之前包的重装下就好。
